content/en/docs/tasks/access-kubernetes-api/custom-resource-definition-versioning.md

```diff
@@ -0,0 +1,238 @@
+- v12alpha1
+- v11alpha2
+- foo1
+- foo10.
+```
+
+For the example in [Specify multiple versions](#specify-multiple-versions), the
+version sort order is `v1`, followed by `v1beta`. This causes the kubectl
+command to use `v1` as the default version unless the provided object specifies
+the version.
+
+## Manage object versions
+
+An object is stored in etcd with the storage version at the time of creation.
+When the object is requested, the API server returns it in the API version of
+the request. In the case of CREATE or UPDATE requests it persists the object
+with the current storage version, not the request version.
+
+To illustrate how this works, you can save the following YAML to
+`my-crontab-v1.yaml`:
+
+```yaml
+apiVersion: "example.com/v1beta1"
+kind: CronTab
+metadata:
+  name: my-new-cron-object
+spec:
+  cronSpec: "* * * * */5"
+  image: my-awesome-cron-image
+```
+
+You can create the `v1beta1` version using the following `kubectl` command.
+
+```shell
+kubectl create -f my-crontab.yaml
+```
+
+The new object is stored as `example.com/v1beta1` in etcd because `v1beta1` is
+the current storage version. If you add a definition for `v1`, it will now be
+first in the sort order, and a newly-created object that doesn't specify an API
+version will be stored as `example.com/v1`.
+
+```shell
+kubectl get my-new-cron-object -o yaml
+```
+
+```yaml
+apiVersion: example.com/v1
+kind: CronTab
+metadata:
+  clusterName: ""
+  creationTimestamp: 2017-05-31T12:56:35Z
+  deletionGracePeriodSeconds: null
+  deletionTimestamp: null
+  name: my-new-cron-object
+  namespace: default
+  resourceVersion: "285"
+  selfLink: /apis/example.com/v1/namespaces/default/crontabs/my-new-cron-object
+  uid: 9423255b-4600-11e7-af6a-28d2447dc82b
+spec:
+  cronSpec: '* * * * */5'
+  image: my-awesome-cron-image
+```
+
+If you specify a version, the object is stored at that version:
+
+```shell
+kubectl get crontabs.v1beta1.stable.example.com my-new-cron-object
+```
+
+```yaml
+apiVersion: example.com/v1beta1
+kind: CronTab
+metadata:
+  clusterName: ""
+  creationTimestamp: 2017-05-31T12:56:35Z
+  deletionGracePeriodSeconds: null
+  deletionTimestamp: null
+  name: my-new-cron-object
+  namespace: default
+  resourceVersion: "285"
+  selfLink: /apis/example.com/v1/namespaces/default/crontabs/my-new-cron-object
+  uid: 9423255b-4600-11e7-af6a-28d2447dc82b
+spec:
+  cronSpec: '* * * * */5'
+  image: my-awesome-cron-image
+```
+
+### Storing multiple versions
+
+You can store different versions of Custom Resources simultaneously. When you
+change the storage version in the CustomResourceDefinition, this only affects
+new and updated objects. Your existing persisted data is not automatically
+upgraded.
+
+The API server records each version which has ever been marked as the storage
+version in the status field `storedVersions`. Expect old versioned objects in etcd for all versions in `status.storedVersions`. Versions may have existed but never been designated as
```

Comment:
"Versions may have existed but never been designated as version": In this case, they will not appear in `storedVersions` field. I think that part of sentence is not required. it can be only: "Versions may designated as the storage version but objects may have never been created for a given version."